bin/templates/cordova/lib/emulator.js

```diff
@@ -227,18 +236,25 @@ module.exports.wait_for_emulator = function(uuid) {
 };
 
 /*
- * Waits for the core android process of the emulator to start
+ * Waits for the core android process of the emulator to start. Returns a
+ * promise that resolves to a boolean indicating success. Not specifying a
+ * time_remaining or passing a negative value will cause it to wait forever
  */
-module.exports.wait_for_boot = function(emulator_id) {
+module.exports.wait_for_boot = function(emulator_id, time_remaining) {
     var self = this;
     return Adb.shell(emulator_id, 'ps')
     .then(function(output) {
         if (output.match(/android\.process\.acore/)) {
-            return;
+            return true;
+        } else if (time_remaining === 0) {
+            return false;
         } else {
             process.stdout.write('.');
-            return Q.delay(3000).then(function() {
-                return self.wait_for_boot(emulator_id);
+
+            // Check at regular intervals
+            return Q.delay(time_remaining < CHECK_BOOTED_INTERVAL ? time_remaining : CHECK_BOOTED_INTERVAL).then(function() {
```

Comment:
Maybe use `Math.min` here instead of the ternary operator? It's a little hard to read long ternary conditions.